File: 12-05-2024/Minimum steps to destination.py
#User function Template for python3

class Solution:
    def minSteps(self, d):
        # code here
        addition = 1
        element = 0
        count = 0
        while element < d:
            element += addition
            addition += 1
            count += 1
        while (element - d) % 2 != 0:
            count += 1
            element += addition
            addition += 1
        return count




#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        d = int(input())

        ob = Solution()
        print(ob.minSteps(d))

# } Driver Code Ends

File: 12-05-2024/Minimum steps to destination.cs
//{ Driver Code Starts
// Initial Template for C#

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DriverCode {

class GFG {
    static void Main(string[] args) {
        int testcases; // Taking testcase as input
        testcases = Convert.ToInt32(Console.ReadLine());
        while (testcases-- > 0) // Looping through all testcases
        {
            int d = Convert.ToInt32(Console.ReadLine());
            Solution obj = new Solution();
            int res = obj.minSteps(d);
            Console.Write(res);
            Console.Write("\n");
        }
    }
}

}

// } Driver Code Ends




// User function Template for C#

class Solution {
    // Complete this function
    public int minSteps(int d) {
        // Your Code Start
        int addition = 1;
        int element = 0;
        int count = 0;
        while (element < d) {
            element += addition;
            addition++;
            count++;
        }
        while ((element - d) % 2 != 0) {
            count++;
            element += addition;
            addition++;
        }
        return count;
      //Your Code End
    }
}
